sphinxcontrib/needs/directives/needtable.py

```diff
@@ -58,6 +60,15 @@ def run(self):
 
         columns = [get_title(col) for col in columns]
 
+        colwidths = str(self.options.get("colwidths", ""))
+        if colwidths:
+            colwidths = [int(width.strip()) for width in re.split(";|,", colwidths)]
+            if len(columns) != len(colwidths):
+                raise NeedsInvalidException(
+                    f"Amount of elements in colwidths and columns do not match: "
+                    f"colwidths: {len(colwidths)} and columns: {len(columns)}"
+                )
+
```

Comment:
can you do this without shadowing the `colwidths` variable with multiple types?

Reply:
Done